src/common/tracer.h

```diff
@@ -0,0 +1,80 @@
+#ifndef TRACER_H_
+#define TRACER_H_
+
+#define SIGNED_RIGHT_SHIFT_IS 1
+#define ARITHMETIC_RIGHT_SHIFT 1
+
+#include<cstring>
+#include <arpa/inet.h>
+#include <yaml-cpp/yaml.h>
+#include<atomic>
+
+#include <jaegertracing/Tracer.h>
+#include <jaegertracing/net/IPAddress.h>
+#include <jaegertracing/net/Socket.h>
+
+
+typedef std::unique_ptr<opentracing::Span> Span;
+
+struct Jaeger_Tracer{
+    Jaeger_Tracer(){}
+    ~Jaeger_Tracer();
+  std::shared_ptr<opentracing::v3::Tracer> tracer = NULL;
+};
+
+//will be used throughout ceph to create spans
+extern Jaeger_Tracer tracer;
```

Comment:
if you want to use a singleton, keep it inside the cc file.
you can wrap it with function, so it does not look like a singleton to the users.
also prevent from anyone changing it in ways that are not desirable.

Reply:
But then it will give me multiple definations error so I chose to declare in .h file

Reply:
declare as static global variable in cc file
do not expose directly in header file
expose functionality via functions